### DISCUSSION TITLE
Is there a possibility to have maintenance windows for airflow scheduler

### DISCUSSION THREAD

**Discussion by waardd** [apache/airflow#37531] [Q&A] [answered]
In our setup we connect to a DB server.
This server gets automated updates every week in the night.

Is it possible to have scheduler and/or webserver in a maintenace window for that.

So set a window for example from saturday 03:00 - 06:00 that no actions are performed from airflow to the database?

**Comment by potiuk** [comment, marked as answer]
There are few options. You can follow this https://airflow.apache.org/docs/apache-airflow/stable/best-practices.html#disable-the-scheduler or 
you can shutdown both for the time period if you want, no problem with that, airlfow should catch-up and reschedule things after that. 

When it comes to shutting down the components, you need to handle it in your deployment to disable monitoring and alerting of course while airflow is shut down, and depending on executor used, it might have more or less impact on the running tasks - generally what you can expect same things when live-upgrading airlfow https://airflow.apache.org/docs/apache-airflow/stable/administration-and-deployment/production-deployment.html#live-upgrading-airflow  

All that however you need to handle as part of your deployment. Airflow as such does not have the concept of maintenance window and you need to implement it outside of Airlfow - using the tools and APIs and mechanisms described in the links above.
